### ISSUE TITLE
Create getter for full attribute path on AbstractPathSpec

### ISSUE THREAD

**Issue by weaselmetal** [crnk-project/crnk-framework#243]
I see this line of code `StringUtils.join(".", attributePath)` many many times in the crnk project code.
Which indicates that a full path String like `person.mother.name` is needed quite often (maybe even more often than the list of path elements, dunno).
So I suggest a getter on `AbstractPathSpec` that implements just that, and spares crnk code and user code to write `StringUtils.join(".", attributePath)` yet another time. Basically just something like

    public String getFullAttributePath() {
      return StringUtils.join(".", attributePath);
    }

I believe that such a getter would also eliminate some complexity in the crnk code base itself.

As an aside, it took me a while to understand why `getAttributePath()` returns a list of strings and what this list is. The getter name indicates that I get 'one path'. That caused some puzzlement on my end at first.

**Comment by remmeier** [comment]
Yes, I think a PathSpec would be a really good thing to have. That list of strings is not overly convenient. But maybe we should also have the generated expression classes already available in Typescript (
http://www.crnk.io/documentation/#_expressions) similar to QueryDSL. This would allow to construct PathSpec without any strings, but in fully type-safe manner.

```
import static QPerson.person;

QuerySpec querySpec = new QuerySpec(person)
querySpec.addFilter(person.mother.name.eq("xy")
...

**Comment by weaselmetal** [comment]
Sounds like a nice idea, but I'd track that as as a different and new idea. Just noted the repeated string join and thought that one obviously wants to work with the full joined path. Also the path comparison and `FilterSpec` retrieval from a `QuerySpec` is less cumbersome.
Something like `person.mother.name.eq("xy")` wouldn't work for our API, really. We map from a String path like "person.mother.name" to a typed filter parameter, in this case that would have to be string value. 
We also need to support some 'made-up' properties that have special semantics on the DB-level.
Something like `person.maxAverageChildAge < 5` where `maxAverageChildAge` doesn't really exist on the entity. One reason why we can't use JpaRepos for example.

**Comment by remmeier** [comment]
BTW: are you aware of the computed attributes in the jpa repositories? depending on how complex things like maxAverageChildAge are, it may would provide a solution.

**Comment by weaselmetal** [comment]
jep, I know those. Thanks for the hint. Another reason why we can't use JpaRepos is that we have a layer between DB entities and exposed objects on the API. No entity is exposed as is on the API. One of the beautiful things you get is:
- name stuff on the API completely different without renaming table columns
- no Jackson annotations on entities
- hopefully easier versioning of the API (haven't had the need yet)

It makes things a little more complicated but the benefits are great.

**Comment by remmeier** [comment]
in some of those casing we make use of the DTO mapping, maybe together with a tool like MapStruct. Not that I want to talk you into JPA repositories :-) But there are quite a few possiblities...

**Comment by weaselmetal** [comment]
We are always open for worthwhile (and non-breaking :P) changes. The DTO mapping would happen on the DB / DAO layer I assume.  Which makes me wonder how one could filter by a property that only exists on a DTO object.

**Comment by remmeier** [comment]
either not filterable, have some kind of napping, are manually added to the query or use the computed attrs.

https://github.com/mapstruct/mapstruct/issues/702#issuecomment-234481663 is something i was looking into
